src/plugin/systems.rs

```diff
@@ -1463,7 +1453,16 @@ pub fn update_character_controls(
                 }
             }
 
-            if let Ok(mut transform) = transforms.get_mut(entity_to_move) {
+            // NOTE: Update the translation of the Rapier's rigid body without editing the Bevy's transform for
+            //       preventing noticing change in the [`systems::apply_rigid_body_user_changes`] for kinematic bodies.
+            if let Some(body) = body_handle.and_then(|h| context.bodies.get_mut(h.0)) {
+                // TODO: Use `set_next_kinematic_translation` for position based kinematic bodies? - Vixenka 29.01.2024
+                body.set_translation(
+                    body.translation() + movement.translation * physics_scale,
```

Comment:
This multiplication is from original line what is lower on line 1468.

Reply:
Not quite. That line refers to the Bevy Transform component. However, this line refers to the *Rapier* body's transform. The Rapier body's transform will get copied to the Bevy Transform later on, where it will be scaled up by the physics scale. So the movement vector gets scaled up by the `physics_scale` twice rather than just once.

I'm not sure if that made sense haha. I tried removing this multiplication on your branch and it works correctly now - Setting my KCC's speed to 50.0 actually produces 50.0 in my `derived_velocity` system.